### ISSUE TITLE
gcsfuse / Google Cloud, big file upload problem 

### ISSUE THREAD

**Issue by skechboy** [jedisct1/pure-ftpd#70]
I'm using pure-ftpd on top on Google Cloud Storage buckets and so far it worked well for files smaller than 1GB. However when we try to upload some larger file around 4GB when the file upload reach 100% the FTP client waits for some time and then shows the following error:

```
Command:	PASV
Response:	227 Entering Passive Mode (146,148,127,80,23,150)
Command:	STOR 4GB-FILE.dat
Response:	150 Accepted data connection
Error:	Connection timed out after 20 seconds of inactivity
Error:	File transfer failed after transferring 4,510,613,504 bytes in 740 seconds
```
However After some time ( 60+s) In pure-ftpd and system log we can see the following: 
```
Nov 22 14:23:02 backup pure-ftpd: (SERVER@XXX.XXX.XXX.XXX) [NOTICE] /home/backup/shared//4GB-FILE.dat uploaded  (4510613504 bytes, 6468.53KB/sec)
SERVER [22/Nov/2017:14:23:02 -0000] "PUT /home/backup/shared/4GB-FILE.dat" 200 4510613504
```
And if we check in Google Cloud storage the file is fully uploaded and it hashed value is correct.

Obviously the issue is timeout related so I'm trying to figure out wither pure-ftpd is having timeout write error because gcsfuse stores the uploaded file locally in a temporary directory and then it flushes it to the Google Cloud bucket storage which depending of the file size can take much more than the 20 seconds showed in the error in the FTP client. 

My question is, if there is some hard coded write / response timeout directive in pure-ftpd which can be updated to suit my needs, or this is a thing that needs to be solved from the client side.

**Comment by jedisct1** [comment]
Hi,

The main thing you can tweak is `MaxIdleTimeMaxIdleTime` but the default is set to 15 minutes, which should be more than enough.

I suspect the connection is actually dropped by the client. Is it something you can tune on your client?

**Comment by skechboy** [comment]
Yep, that is correct the connection is dropped by the client because the server isn't returning any response during this period. Updating client timeout setting is solving the issue currently but it may be problem when
